modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/display_layout/edit_asset_category_cp_display_layout.jsp

```diff
@@ -99,7 +99,7 @@ if (cpDisplayLayout != null) {
 	</aui:form>
 </commerce-ui:side-panel-content>
 
-<aui:script use="liferay-item-selector-dialog">
+<aui:script>
```

Comment:
How Do you balance between using `liferay-frontend:component` tag and just inline on JSP using `aui:script`?

Considering the size of this JS code on JSP, can We consider creating a JS file for it?

Reply:
> How Do you balance between using liferay-frontend:component tag and just inline on JSP using aui:script?

I don't know, I haven't considered changing it into `liferay-frontent:component` or a separate file. I guess it can be done, yea, although maybe not considering how buggy Commerce is. Thoughts @markocikos?

Reply:
Obviously it's a judgment call. There is a cost to moving stuff into a separate file — namely, you have a separate file. 😂 But the upside is potentially quite large: you get a "proper" JS environment with no arbitrary limitation (ie. testing, linting, formatting, tranpilation, all work using our standard/modern stack, and you probably get a better editing experience too). You can also use TypeScript in stand-alone files, something that I expect we're never going to be able to do inside JSP. And finally, if you use `liferay-frontend:component` you get some "lifecycle management" for free, in addition to making the world a better place by keeping the JS in JSP to a minimum.

> although maybe not considering how buggy Commerce is

Bugginess probably isn't an important input into this decision.

Reply:
> Bugginess probably isn't an important input into this decision.

I'm convinced, since this PR is probably gonna be in draft for a little bit longer I'll mess about with it and try to leave the world a better place than I found it 👍 

P.S. Love you @wincent